lib/services/ios-log-filter.ts

```diff
@@ -1,65 +1,86 @@
 const sourcemap = require("source-map");
 import * as path from "path";
 import { cache } from "../common/decorators";
-import * as iOSLogFilterBase from "../common/mobile/ios/ios-log-filter";
 
-export class IOSLogFilter extends iOSLogFilterBase.IOSLogFilter implements Mobile.IPlatformLogFilter {
-	protected infoFilterRegex = /^.*?((?:<Notice>:)?.*?(((?:CONSOLE|JS) (?:LOG|ERROR)).*?))$/im;
+export class IOSLogFilter implements Mobile.IPlatformLogFilter {
+	// Used to recognize output related to the current project
+	// This looks for artifacts like: AppName[22432] or AppName(SomeTextHere)[23123]
+	private appOutputRegex: RegExp = /([^\s\(\)]+)(?:\([^\s]+\))?\[[0-9]+\]/;
+
+	// Used to trim the passed messages to a simpler output
+	// Example:
+	// This: "May 24 15:54:52 Dragons-iPhone NativeScript250(NativeScript)[356] <Notice>: CONSOLE ERROR file:///app/tns_modules/@angular/core/bundles/core.umd.js:3477:36: ORIGINAL STACKTRACE:"
+	// Becomes: CONSOLE ERROR file:///app/tns_modules/@angular/core/bundles/core.umd.js:3477:36: ORIGINAL STACKTRACE:
+	protected infoFilterRegex =  new RegExp(`^.*(?:<Notice>:|<Error>:|<Warning>:|\\(NativeScript\\)|${this.appOutputRegex.source}:){1}`);
+
+	private filterActive: boolean = true;
+	private projectName: string;
 
 	private partialLine: string = null;
 
-	constructor($loggingLevels: Mobile.ILoggingLevels,
+	constructor(
+		private $loggingLevels: Mobile.ILoggingLevels,
 		private $fs: IFileSystem,
 		private $projectData: IProjectData) {
-		super($loggingLevels);
+		this.projectName = this.$projectData.projectName;
```

Comment:
I think this might not work in Sidekick as there only one instance is created of the filter. Try getting projectData from projectDataService.getProjectData and using the project name from there on evry invocation of the filterData function or pass it as an argument if available where the method gets called.